File: src/main/java/org/jetbrains/pmdkotlin/lang/kotlin/KotlinHandler.java
package org.jetbrains.pmdkotlin.lang.kotlin;

import net.sourceforge.pmd.lang.AbstractLanguageVersionHandler;
import net.sourceforge.pmd.lang.Parser;
import net.sourceforge.pmd.lang.ParserOptions;
import net.sourceforge.pmd.lang.rule.RuleViolationFactory;
import org.jetbrains.pmdkotlin.lang.kotlin.rule.KotlinRuleViolationFactory;

public class KotlinHandler extends AbstractLanguageVersionHandler {

    @Override
    public RuleViolationFactory getRuleViolationFactory() {
        return KotlinRuleViolationFactory.INSTANCE;
    }

    @Override
    public Parser getParser(ParserOptions parserOptions) {
        return new KotlinParser(parserOptions);
    }
}

File: src/main/kotlin/org/jetbrains/pmdkotlin/lang/kotlin/KotlinHandler.kt
package org.jetbrains.pmdkotlin.lang.kotlin

import net.sourceforge.pmd.lang.AbstractLanguageVersionHandler
import net.sourceforge.pmd.lang.Parser
import net.sourceforge.pmd.lang.ParserOptions
import net.sourceforge.pmd.lang.rule.RuleViolationFactory
import org.jetbrains.pmdkotlin.lang.kotlin.rule.KotlinRuleViolationFactory

class KotlinHandler : AbstractLanguageVersionHandler() {

    override fun getRuleViolationFactory(): RuleViolationFactory {
        return KotlinRuleViolationFactory.INSTANCE
    }

    override fun getParser(parserOptions: ParserOptions): Parser {
        return KotlinParser(parserOptions)
    }
}
